src/findlib/findlib.ml

```diff
@@ -109,9 +115,31 @@ let command_names cmd_spec =
 let auto_config_file() =
   let p =
     ( try Sys.getenv "OCAMLFIND_CONF" with Not_found -> "") in
-  if p = "" then Findlib_config.config_file else p
+  if p = "" then Lazy.force Findlib_config.config_file else p
+
+let relocatable_path path =
+  let prefix = "$PREFIX" in
+  let len = String.length prefix in
+  match String.starts_with ~prefix path with
+  | false -> None
+  | true -> Some (String.sub path len (String.length path - len))
```

Comment:
any special handling of `/` needed? E.g.
 - `prefix="/d1/d2"` and `path="/d1/d2goeson"`: returns `goeson`
 - `prefix="/d1/d2"` and `path="/d1/d2/d3/foo"`: returns `/d3/foo` (absolute!)

Reply:
Not in this function because it is should indeed return that absolute path (I've renamed it since), so returning the absolute part should not be a problem, given it is later only used to `Filename.concat` with the computed prefix.

However with that concern in mind, if the path was `$PREFIX/d3/foo` and we couldn't compute a `Findlib.location` it would wrongly add `/d3/foo` to the search path. I've changed it to (silently) discard the path in such case. The alternative would be to expand $PREFIX/d3/foo` to a relative `d3/foo` but that behavior seems a bit surprising and unexpected.